File: src/main/java/org/hwabeag/hwaskyblock/schedules/UnloadTask.java
package org.hwabeag.hwaskyblock.schedules;

import org.bukkit.Bukkit;
import org.bukkit.World;

import java.io.File;

public class UnloadTask implements Runnable {

    @Override
    public void run() {
        for (World world : Bukkit.getServer().getWorlds()) {
            String worldName = world.getName();
            if (worldName.contains("HwaSkyBlock.")) {
                File worldDir = new File(Bukkit.getServer().getWorldContainer(), worldName);

                if (!worldDir.exists()) {
                    Bukkit.getLogger().warning("World directory for " + worldName + " does not exist.");
                    continue;
                }

                if (world.getPlayers().isEmpty()) {
                    try {
                        Bukkit.getServer().unloadWorld(world, true);
                        Bukkit.getLogger().info("Successfully unloaded world: " + worldName);
                    } catch (Exception e) {
                        Bukkit.getLogger().severe("Failed to unload world: " + worldName);
                        e.printStackTrace();
                    }
                }
            }
        }
    }
}

File: src/main/kotlin/schedules/UnloadTask.kt
package org.hwabaeg.hwaskyblock.schedules

import org.bukkit.Bukkit
import java.io.File

class UnloadTask : Runnable {
    override fun run() {
        for (world in Bukkit.getServer().worlds) {
            val worldName = world.name
            if (worldName.contains("HwaSkyBlock.")) {
                val worldDir = File(Bukkit.getServer().worldContainer, worldName)

                if (!worldDir.exists()) {
                    Bukkit.getLogger().warning(worldName + "에 대한 세계 디렉토리가 존재하지 않습니다.")
                    continue
                }

                if (world.players.isEmpty()) {
                    try {
                        Bukkit.getServer().unloadWorld(world, true)
                        Bukkit.getLogger().info("세계를 성공적으로 언로드했습니다: $worldName")
                    } catch (e: Exception) {
                        Bukkit.getLogger().severe("세계 언로드 실패: $worldName")
                        e.printStackTrace()
                    }
                }
            }
        }
    }
}